tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -408,16 +407,21 @@ SpanBuffer SwapBuffers()
                     }
                 }
 
-                if (!shouldSendTrace)
+                if (shouldSendTrace)
+                {
+                    TelemetryFactory.Metrics.RecordCountTraceChunkEnqueued(MetricTags.TraceChunkEnqueueReason.P0Keep);
+                }
+                else
                 {
                     // If stats computation determined that we can drop the P0 Trace,
                     // skip all other processing
+                    TelemetryFactory.Metrics.RecordCountTraceChunkDropped(MetricTags.DropReason.P0Drop);
                     if (singleSpanSamplingSpans.Count == 0)
                     {
                         Interlocked.Increment(ref _droppedP0Traces);
                         Interlocked.Add(ref _droppedP0Spans, spans.Count);
-                        TelemetryFactory.Metrics.RecordCountTraceDropped(MetricTags.DropReason.SamplingDecision);
-                        TelemetryFactory.Metrics.RecordCountSpanDropped(MetricTags.DropReason.SamplingDecision, spans.Count);
+                        TelemetryFactory.Metrics.RecordCountTraceChunkDropped(MetricTags.DropReason.P0Drop);
```

Comment:
I believe we should remove this call as it is counted before the if as well and as in the else we increment `_droppedP0Traces` so I assume we want to account this in both cases.

Reply:
Yep you're right, thanks!